app/lib/search_service.dart

```diff
@@ -78,19 +78,14 @@ class SearchService {
 
       if (keys.isNotEmpty) {
         final List<Package> packages = await db.lookup(keys);
+        packages.removeWhere((p) => !exists(p));
         final List<Key> versionKeys =
-            packages.where(exists).map((p) => p.latestVersionKey).toList();
+            packages.map((p) => p.latestVersionKey).toList();
+        final List<Key> devVersionKeys =
+            packages.map((p) => p.latestDevVersionKey).toList();
         if (versionKeys.isNotEmpty) {
-          // select latest development versions for each package
-          final packageKeys = versionKeys.map((vk) => vk.parent).toList();
-          final List<PackageVersion> devVersions =
-              await Future.wait(packageKeys.map((p) async {
-            final List<PackageVersion> all =
-                await db.query(PackageVersion, ancestorKey: p).run().toList();
-            all.sort((a, b) => b.semanticVersion.compareTo(a.semanticVersion));
-            return all.first;
-          }));
           final versions = await db.lookup(versionKeys);
+          final devVersions = await db.lookup(devVersionKeys);
```

Comment:
Fixed. I went with the first version, because most dev versions will be the same as the stable version, and I hope that the DataStore will figure that out and won't do duplicate reads.